src/zkevm_specs/evm/execution/slt_sgt.py

```diff
@@ -0,0 +1,63 @@
+from typing import Sequence, Tuple
+
+from ..instruction import Instruction, Transition
+from ..opcode import Opcode
+
+
+def scmp(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+
+    is_sgt, _ = instruction.pair_select(opcode, Opcode.SGT, Opcode.SLT)
+
+    a = instruction.stack_pop()
+    b = instruction.stack_pop()
+    c = instruction.stack_push()
+
+    # decode RLC to bytes for a and b
+    # Note: `rlc_to_bytes` returns the bytes in big endian form. In order to
+    # match the specs and circuit implementation, we transform them to
+    # little endian form (which is the default form returned by `query_word`
+    a8s = instruction.rlc_to_bytes(a, 32)
+    b8s = instruction.rlc_to_bytes(b, 32)
+    c8s = instruction.rlc_to_bytes(c, 32)
+    a8s.reverse()
+    b8s.reverse()
+    c8s.reverse()
+
+    cc = int.from_bytes(c8s, "little")
+    a_hi = int.from_bytes(a8s[16:32], "little")
+    a_lo = int.from_bytes(a8s[0:16], "little")
+    b_hi = int.from_bytes(b8s[16:32], "little")
+    b_lo = int.from_bytes(b8s[0:16], "little")
+
+    a_lt_b = (a_hi < b_hi) or ((a_hi == b_hi) and (a_lo < b_lo))
```

Comment:
Just one confirm:  if a =0b10.......1 (-1), b = 0b10....0  (1) in little endian form,   does the `a_hi > b_hi ` by `a_hi = int.from_bytes(a8s[16:32],  b_hi = int.from_bytes(b8s[16:32], "little")` "little")?

Others good to me !

Reply:
`a` and `b` are both in the two's complement form. So `-1` would be: `0b1111...1111` and `1` would be `0b1000...0000` (both little endian representations).

And if `a < 0` and `b >= 0`, we return `a < b -> 1`. We compare `a_hi`, `b_hi`, `a_lo` and `b_lo` only if both `a` and `b` have the same sign, i.e. `a < 0 && b < 0` or `a >= 0 && b >= 0`